File: checkpoint0/src/test/java/com/springernature/checkpoint0/BoardTest.java
package com.springernature.checkpoint0;

import org.junit.Test;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.IntStream;

import static com.springernature.checkpoint0.Cell.Empty;
import static com.springernature.checkpoint0.Cell.X;
import static java.util.Arrays.asList;
import static java.util.stream.Collectors.toList;
import static org.hamcrest.core.IsEqual.equalTo;
import static org.junit.Assert.assertThat;

public class BoardTest {

    @Test
    public void setAValueOnTheBoard() {
        List<List<Cell>> lists = IntStream.range(0, 3)
                .mapToObj(index -> new ArrayList<>(asList(Empty, Empty, Empty)))
                .collect(toList());
        Board board = new Board(lists);

        Board updatedBoard = board.set(0, 1, X);

        assertThat(updatedBoard.get(0, 1), equalTo(X));
    }
}

File: checkpoint1/src/test/kotlin/com/springernature/checkpoint1/BoardTest.kt
package com.springernature.checkpoint1

import com.springernature.checkpoint1.Cell.Empty
import com.springernature.checkpoint1.Cell.X
import org.hamcrest.core.IsEqual.equalTo
import org.junit.Assert.assertThat
import org.junit.Test

class BoardTest {

    @Test
    fun `set a value on the board`() {
        val lists = 0.until(3).map { List(3, { Empty }) }

        val board = Board(lists)
        val actual = board.set(0, 1, X)

        assertThat(actual[0, 1], equalTo(X))
    }
}